File: examples/org.specapi.plugins.dotnet.example/dotnet-gen/Org.Specapi.Github/ListAllPublicRepositoriesRequest.cs
using System.Collections.Specialized;
using System.Linq;
using System;
using System.Net;
namespace Org.Specapi.Github
{
    public partial class ListAllPublicRepositoriesRequest
    {
        bool _sinceParamSet = false;
        string _sinceParam;
        public string Since 
        { 
            get 
            {
                return _sinceParam;
            } 
            set 
            {
                _sinceParamSet = true;
                _sinceParam = value;
            }
        }
        
        public delegate void Status200Handler(Repository[] result);
        public delegate void StatusUnexpectedHandler(HttpWebResponse result);
        public Status200Handler On200 { get; set; }
        public StatusUnexpectedHandler OnOther { get; set; }
                
        public ListAllPublicRepositoriesRequest() {
        }
        
        public Uri CreateUri (string baseUrl)
        {
            String url = baseUrl + "/repositories";
            
            var uriBuilder = new UriBuilder(url);
            NameValueCollection query = new NameValueCollection ();
            if(_sinceParamSet) {
                query ["since"] = _sinceParam;
            }
            uriBuilder.Query = string.Join("&", query.AllKeys.Select(key => string.Format("{0}={1}", WebUtility.UrlEncode(key), WebUtility.UrlEncode(query[key]))));
            return uriBuilder.Uri;
        }
    }
}

File: examples/org.specapi.plugins.android.example/src-gen/org/specapi/github/ListAllPublicRepositoriesRequest.java
package org.specapi.github;


import android.net.Uri;
import org.specapi.android.net.ServiceRequest;

public class ListAllPublicRepositoriesRequest extends ServiceRequest {
	
	private static final String PATH = "/repositories";
	
	private String sinceParam;
	private boolean sinceParamSet;
		
	public ListAllPublicRepositoriesRequest setSinceParam(String value) {
		this.sinceParam = value;
		this.sinceParamSet = true;
		return this;
	}
	
	public boolean isSinceParamSet() {
		return sinceParamSet;
	}
		
	public ListAllPublicRepositoriesRequest(){
	}
	
	@Override
	public String createUrl(String baseUrl){
		Uri.Builder uriBuilder = Uri.parse(baseUrl + PATH).buildUpon();
			
		if(sinceParamSet){
			uriBuilder.appendQueryParameter("since", sinceParam);
		}
		
		return uriBuilder.toString();			
	}
}
